staging/src/k8s.io/api/batch/v1/types.go

```diff
@@ -375,6 +375,11 @@ type CronJobSpec struct {
 	// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
 	Schedule string `json:"schedule" protobuf:"bytes,1,opt,name=schedule"`
 
+	// The time zone for the given schedule, see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones.
```

Comment:
how stable is the list of time zones. are new allowed values added in new go versions or OS updates? are existing valid values ever dropped? if we're adopting this in API validation, we need to make sure we don't put ourselves in a position where two skewed versions of the API server running against the same etcd simultaneously disagree about what a valid time zone is

Reply:
From what I've seen, zones that are "dropped" are usually aliased to the more currently accurate zone. I did add more tests to confirm that the time zones are case-sensitive.

Reply:
good to see the added unit tests. compatibility implications of this field over time are important to resolve before merge. There are two aspects:

1) API validation not rejecting a value in this field that was previously allowed to be persisted... that means that when validating a cronjob API object update, if this field doesn't change, we shouldn't forbid the write.

2) how the controller handles locations that cannot be parsed... currently it is defaulting to local time - there's an outstanding question on the controller about whether it is better to run in the wrong time zone or not at all

Reply:
Looks like the controller case is now handled. Behavior of the API on an update of a cronjob whose existing timezone is considered invalid is not yet handled